src/confluent_kafka/aio/_AIOProducer.py

```diff
@@ -21,36 +21,102 @@
+        wrap_common_callbacks(asyncio.get_running_loop(), producer_conf)
 
         self._producer = confluent_kafka.Producer(producer_conf)
-        self._running = False
-        if auto_poll:
-            self._running = True
-            self._running_loop = asyncio.create_task(self._loop())
-
-    async def stop(self):
-        if self._running:
-            self._running = False
-            await self._running_loop
-
-    async def _loop(self):
-        while self._running:
-            await self.poll(1.0)
+        
+        # Internal metrics tracking
+        self._metrics_callback = None
+        self._send_times = {}  # Track send times for latency calculation
+
+    def set_metrics_callback(self, callback):
+        """Set callback for delivery metrics: callback(latency_ms, topic, partition, success)"""
+        self._metrics_callback = callback
+
+    # ========================================================================
+    # HYBRID OPERATIONS - Blocking behavior depends on parameters
+    # ========================================================================
+
+    async def poll(self, timeout=0, *args, **kwargs):
+        """Processes callbacks - blocking behavior depends on timeout
+        
+        Args:
+            timeout: 0 = non-blocking, >0 = block up to timeout seconds, -1 = block indefinitely
+        """
+        if timeout > 0:
+            # Blocking call - use ThreadPool to avoid blocking event loop
+            return await self._call(self._producer.poll, timeout, *args, **kwargs)
+        else:
+            # Non-blocking call (timeout=0) - direct call is safe
+            return self._producer.poll(timeout, *args, **kwargs)
+
+    # ========================================================================
+    # NON-BLOCKING OPERATIONS - Direct calls (no ThreadPool overhead)
+    # These operations are already async-safe in librdkafka
+    # ========================================================================
+
+    async def produce(self, topic, value=None, key=None, *args, **kwargs):
+        """Non-blocking: Queues message and returns immediately
+        
+        Args:
+            topic: Kafka topic name (required)
+            value: Message payload (optional)
+            key: Message key (optional)
+            *args, **kwargs: Additional parameters like partition, timestamp, headers
+        """
+        # Get current running event loop
+        result = asyncio.get_running_loop().create_future()
+        
+        # Track send time for latency calculation if metrics are enabled
+        send_time = None
+        message_id = None
+        if self._metrics_callback:
+            import time
+            send_time = time.time()
+            message_id = f"{topic}:{key}:{id(result)}"  # Unique message identifier
+            self._send_times[message_id] = send_time
+
+        # Pre-bind variables to avoid closure overhead
+        def on_delivery(err, msg):
```

Comment:
My knee jerk reaction for a tight loop call was to ask to avoid doing a function generation within that loop. But that code path has been heavily optimized since Python 2 days (and I checked that `lambda` has a very similar low cost profile without any perf differences).